Source/SwiftLintFramework/Rules/SubClassRule.swift

```diff
@@ -0,0 +1,47 @@
+//
+//  SubClassRule.swift
+//  SwiftLint
+//
+//  Created by Mikhail Yakushin on 03/02/18.
+//  Copyright © 2018 Realm. All rights reserved.
+//
+
+import SourceKittenFramework
+
+public struct SubClassRule: ASTRule, ConfigurationProviderRule, OptInRule {
+    public var configuration = SeverityLevelsConfiguration(warning: 0, error: 0)
+
+    public init() {}
+
+    public static let description = RuleDescription(
+        identifier: "subclass",
+        name: "Subclass",
+        description: "Subclassing is prohibited.",
+        kind: .style
+    )
+
+    public func validate(file: File, kind: SwiftDeclarationKind,
+                         dictionary: [String: SourceKitRepresentable]) -> [StyleViolation] {
+        guard SwiftDeclarationKind.functionKinds.contains(kind),
+              let offset = dictionary.offset,
+              case let contentsNSString = file.contents.bridge()
+            else {
+            return []
+        }
+
+        if contentsNSString.contains("super.") || contentsNSString.contains("super()") {
```

Comment:
this is not the right way of doing this since you can have subclasses that don't call `super`

Reply:
@marcelofabri What would be the right way in your opinion? The `SourceKit` does not distinguish the inheritance between the interfaces and superclasses. It only shows that the class inherits from something.

Reply:
IMO we should have a rule that triggers if you don't add `final` to a class, as @SDGGiesbrecht [suggested](https://github.com/realm/SwiftLint/issues/2079#issuecomment-371944096).

There's not a lot of value (and in fact I'd argue that it's worst) to trigger a violation on cases where you don't have an alternative, usually dealing with dependencies (either system frameworks like UIKit or 3rd party ones).